src/components/CustomTable/CustomTable.tsx

```diff
@@ -0,0 +1,143 @@
+import { Box, Table, TableBody, TableCell, TableContainer, TablePagination, TableRow } from '@mui/material';
+import { ChangeEvent, MouseEvent, useEffect } from 'react';
+import Checkbox from '@mui/material/Checkbox';
+import { CustomTableHead } from '../CustomTableHead/CustomTableHead';
+import { CustomTableProps } from './CustomTable.types';
+import { FetchError } from '../../errors/FetchError';
+
+export function CustomTable<T extends Record<PropertyKey, unknown>>(props: CustomTableProps<T>) {
+  const { toolbar, fetchUrl, selectedRowsIds, page, rows, rowsPerPage, setSelectedRowsIds, setPage, setRowsPerPage, setRows } = props;
+
+  const isSelected = (row: T) => {
+    return (
+      selectedRowsIds.filter((item) => {
+        return (row.id as string) === item;
+      }).length > 0
+    );
+  };
+
+  const pixelHeightPerRow = 53;
+  const rowsPerPageArray = [5, 10, 25];
+
+  const handleClick = (event: MouseEvent<unknown>, row: T) => {
+    const selectedIndex = selectedRowsIds.indexOf(
+      selectedRowsIds.find((item) => {
+        return String(row.id) === item;
+      }) as string
+    );
+    let newSelectedRowsIds: readonly string[];
+
+    if (selectedIndex === -1) {
+      newSelectedRowsIds = [...selectedRowsIds, String(row.id)];
+    } else {
+      const firstPart = selectedRowsIds.slice(0, selectedIndex);
+      const secondPart = selectedRowsIds.slice(selectedIndex + 1);
+      newSelectedRowsIds = [...firstPart, ...secondPart];
+    }
+
+    setSelectedRowsIds(newSelectedRowsIds);
+  };
+
+  const handleChangePage = (_event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event: ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  const emptyRows = page > 0 ? Math.max(0, (1 + page) * rowsPerPage - rows.length) : 0;
+
+  useEffect(() => {
+    fetch(fetchUrl)
+      .then((res) => {
+        return res.json();
+      })
+      .then((data: T[]) => {
+        return setRows(data);
+      })
+      .catch((error) => {
+        if (error instanceof FetchError) {
+          throw new FetchError(`Failed to fetch with error: ${error.message}`);
+        }
```

Comment:
What if this is a different type of error? Just do one check for all type of error for rejected promise